pkgs/development/tools/swiftpm2nix/swiftpm2nix.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+# Generates a Nix expression to fetch swiftpm dependencies, and a
+# configurePhase snippet to prepare a working directory for swift-build.
+
+set -eu -o pipefail
+shopt -s lastpipe
+
+pinFile="Package.resolved"
+stateFile=".build/workspace-state.json"
+
+for file in $pinFile $stateFile; do
+  if [[ ! -f "$file" ]]; then
+    echo >&2 "Missing $file. Run 'swift package resolve' first."
+    exit 1
+  fi
+done
+
+if [[ "$(jq .version $pinFile)" != "1" ]]; then
+  echo >&2 "Unsupported $pinFile version"
+  exit 1
+fi
+if [[ "$(jq .version $stateFile)" != "5" ]]; then
+  echo >&2 "Unsupported $stateFile version"
+  exit 1
+fi
+
+# Iterate dependencies and prefetch.
+sources=""
+jq -r '.object.dependencies[] | "\(.subpath) \(.packageRef.location) \(.state.checkoutState.revision)"' $stateFile \
```

Comment:
As in `jq nix-prefetch-git`? I don't think that's necessary because nix provides those. Most scripts in nixpkgs use a nix-shell shebang, but in this case it's an installable (will work with `nix-shell -p swiftpm2nix`) wrapped to have those dependencies available.